apps/mobile/src/components/ChatDetailScreen/components/ImageZoomOverlay.tsx

```diff
@@ -0,0 +1,109 @@
+import {BackHandler, Modal, Platform, useWindowDimensions} from 'react-native'
+import {useAtom} from 'jotai'
+import {Stack, XStack} from 'tamagui'
+import {useSafeAreaInsets} from 'react-native-safe-area-context'
+import IconButton from '../../IconButton'
+import closeSvg from '../../images/closeSvg'
+import {ImageZoom} from '@likashefqet/react-native-image-zoom'
+import {gestureHandlerRootHOC} from 'react-native-gesture-handler'
+import {useFocusEffect} from '@react-navigation/native'
+import React from 'react'
+import {useMolecule} from 'jotai-molecules'
+import * as Sharing from 'expo-sharing'
+import * as E from 'fp-ts/Either'
+import {chatMolecule} from '../atoms'
+import shareIosSvg from '../images/shareIosSvg'
+import shareAndroidSvg from '../images/shareAndroidSvg'
+import {toError} from '@vexl-next/domain/dist/utility/errors'
+
+const ZoomedImage = gestureHandlerRootHOC(() => {
+  const {width} = useWindowDimensions()
+  const {bottom, top, right, left} = useSafeAreaInsets()
+  const {openedImageUriAtom} = useMolecule(chatMolecule)
+  const [openedImageUri, setOpenedImageUri] = useAtom(openedImageUriAtom)
+
+  async function shareImage(): Promise<void> {
```

Comment:
It seems wrong to use this without useCallback. But I also don't see any props/hooks that might cause this to rerender (and trigger unnecessary rerendering). But I would still prefere to have this in useCallback. 

Is there any reason why is this not in useCallback?